src/main/java/ru/my/Mocito/departmentService/DepartmentImpl.java

```diff
@@ -0,0 +1,54 @@
+package ru.my.Mocito.departmentService;
+import org.springframework.stereotype.Service;
+import ru.my.Mocito.Employee;
+import ru.my.Mocito.employeeService.Iemployee;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class DepartmentImpl implements Idepartment {
+    private final Iemployee iemployee;
+
+    public DepartmentImpl(Iemployee iemployee) {
+        this.iemployee = iemployee;
+    }
+
+    @Override
+    public Map<Integer, List<Employee>> depaertments() {
+        return iemployee.findAll().stream()
+                .collect(Collectors.groupingBy(Employee::getDepartment));
+    }
+
+    @Override
+    public ArrayList<Employee> employees(int id) {
```

Comment:
```suggestion
    public List<Employee> employees(int id) {
```
надо завязываться на абстракции, а не реализации